src/test-run/index.ts

```diff
@@ -1236,7 +1236,7 @@ export default class TestRun extends AsyncEventEmitter {
         const { screenshots } = this.opts;
 
         if (!this.errScreenshotPath && (screenshots as ScreenshotOptionValue)?.takeOnFails)
-            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId })) as string;
+            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: (screenshots as ScreenshotOptionValue)?.fullPage })) as string;
```

Comment:
Please make an assignment of `screenshots` in this way `const { screenshots } = this.opts as { screenshots: ScreenshotOptionValue };`. It'll allow you to escape from the repeating `(screenshots as ScreenshotOptionValue)`.